client/web/compose/src/components/PageBlocks/RecordListBase.vue

```diff
@@ -1168,7 +1172,51 @@ export default {
       this.processing = false
     },
 
-    handleRowClicked ({ r: { recordID } }) {
+    handleRowClicked ({ r: { recordID, values, ...record } }) {
+      const { moduleID, namespaceID } = this.recordListModule
+      let prevAndNextRecordFilters = []
+      const field = this.options.presort.split(' ')[0] || 'createdAt'
+      const value = values[field] || record[field]
+
+      const moduleField = this.recordListModule.findField(field)
+
+      prevAndNextRecordFilters.push({
+        condition: 'Where',
+        name: moduleField.name,
+        operator: '>=',
+        value: value,
+        kind: moduleField.kind,
+      })
+
+      prevAndNextRecordFilters.push({
+        condition: 'Where',
+        name: moduleField.name,
+        operator: '<',
+        value: value,
+        kind: moduleField.kind,
+      })
+
+      const query = queryToFilter(this.query, this.prefilter, this.recordListModule.filterFields(this.options.fields), this.recordListFilter)
+
+      prevAndNextRecordFilters = prevAndNextRecordFilters.map(filter => getFieldFilter(filter.name, filter.kind, filter.value, filter.operator))
```

Comment:
Instead of maping , just use this getFieldFilter when you define the initial array, and provide (name, kind, value, operator)